File: superheroconnections/src/main/java/com/lawrence/superheroconnections/ConnectionActivity.java
package com.lawrence.superheroconnections;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.lawrence.superheroapp.rest.RetrofitClient;
import com.lawrence.superheroapp.rest.model.Connections;
import com.lawrence.superheroapp.rest.model.Id;
import com.lawrence.superheroapp.rest.network.GetService;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ConnectionActivity extends AppCompatActivity {

    private EditText mSearchconnection;
    private TextView mName, mGroup, mRelatives;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_connection);

        mSearchconnection = findViewById(R.id.search_connections);
        mName = findViewById(R.id.connection_name);
        mGroup = findViewById(R.id.connection_group);
        mRelatives = findViewById(R.id.connection_relatives);


    }
    public void Search(View view) {
        final String characterId = mSearchconnection.getText().toString().trim();
            GetService service = RetrofitClient.buildService(GetService.class);
            Call<Connections> call = service.getConnections(characterId);
            call.enqueue(new Callback<Connections>() {
                @Override
                public void onResponse(Call<Connections> call, Response<Connections> response) {
                    mSearchconnection.setVisibility(View.INVISIBLE);
                    mName.setText(String.format("Name: %s", response.body().getName()));
                    mGroup.setText(String.format("Group: %s", response.body().getGroupAffiliation()));
                    mRelatives.setText(String.format("Relatives: %s", response.body().getRelatives()));

                }

                @Override
                public void onFailure(Call<Connections> call, Throwable t) {
                    mSearchconnection.setVisibility(View.INVISIBLE);
                    Toast.makeText(ConnectionActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_LONG).show();
                }
            });
    }
}

File: superheroconnections/src/main/java/com/lawrence/superheroconnections/ConnectionActivity.kt
package com.lawrence.superheroconnections

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.lawrence.superheroapp.rest.RetrofitClient
import com.lawrence.superheroapp.rest.model.Connections
import com.lawrence.superheroapp.rest.network.GetService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ConnectionActivity : AppCompatActivity() {
    private var mSearchconnection: EditText? = null
    private var mName: TextView? = null
    private var mGroup: TextView? = null
    private var mRelatives: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_connection)
        mSearchconnection = findViewById(R.id.search_connections)
        mName = findViewById(R.id.connection_name)
        mGroup = findViewById(R.id.connection_group)
        mRelatives = findViewById(R.id.connection_relatives)
    }

    fun Search(view: View?) {
        val characterId = mSearchconnection!!.text.toString().trim { it <= ' ' }
        val service = RetrofitClient.buildService(GetService::class.java)
        val call = service.getConnections(characterId)
        call?.enqueue(object : Callback<Connections> {
            override fun onResponse(call: Call<Connections>, response: Response<Connections>) {
                mSearchconnection!!.visibility = View.INVISIBLE
                mName!!.text = String.format("Name: %s", response.body()!!.name)
                mGroup!!.text = String.format("Group: %s", response.body()!!.groupAffiliation)
                mRelatives!!.text = String.format("Relatives: %s", response.body()!!.relatives)
            }

            override fun onFailure(call: Call<Connections>, t: Throwable) {
                mSearchconnection!!.visibility = View.INVISIBLE
                Toast.makeText(this@ConnectionActivity, "Error: " + t.message, Toast.LENGTH_LONG).show()
            }
        })
    }
}

private fun <T> Call<T>?.enqueue(callback: Callback<Connections>) {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
}
